source/gateway.go

```diff
@@ -294,18 +296,33 @@ func (c *gatewayRouteResolver) resolve(rt gatewayRoute) (map[string]endpoint.Tar
 	}
 	hostTargets := make(map[string]endpoint.Targets)
 
+	routeParentRefs := rt.ParentRefs()
+
+	if len(routeParentRefs) == 0 {
+		log.Debugf("No parent references found for %s %s/%s", c.src.rtKind, rt.Metadata().Namespace, rt.Metadata().Name)
+		return hostTargets, nil
+	}
+
 	meta := rt.Metadata()
 	for _, rps := range rt.RouteStatus().Parents {
 		// Confirm the Parent is the standard Gateway kind.
 		ref := rps.ParentRef
+		namespace := strVal((*string)(ref.Namespace), meta.Namespace)
+		// Ensure that the parent reference is in the routeParentRefs list
+		hasParentRef := gwRouteHasParentRef(routeParentRefs, ref, meta)
+
+		if !hasParentRef {
```

Comment:
Sorry, this is super nit picky, but can you get rid of the new variable?

e.g.
```go
		// Ensure that the parent reference is in the routeParentRefs list
		if !gwRouteHasParentRef(routeParentRefs, ref, meta) {
```

Reply:
No problem that makes sense. Thanks!